FNML/ontology.md

```diff
@@ -56,24 +55,42 @@ It is still undecided whether this extension should stay (so `rr:Literal` by def
 or go (so `rr:IRI` by default).
 </p>
 
+An [=fnml:ExecutionTermMap=] MUST have exactly one `fnml:execution` relation.
+Further, it MAY have following relations specified:
+
+* `rr:termType`: for processing, see paragraph above
+* `rr:language` OR `rml:langaugeMap` OR `rr:datatype`: for processing, see [RML Language Tags](https://rml.io/specs/rml/#language-tag) and [RML Typed Literals](https://rml.io/specs/rml/#typed-literals)
+* `fnml:output`: this relationship MUST refer to exactly one of the output predicates as specified in the FnO [=function description=]. This signifies which result of the execution to use. The default value is the first output predicate as specified in the FnO [=function description=]. This order is deterministic as the outputs of an FnO [=function description=] are described in an rdf:List.
+
 <p class="issue" data-number="7" data-format="markdown">
-It is currently unspecified how to override the termtype of a <a>function map</a> result.
+Is it still unspecified how to override the termtype of an [=execution term map=] result?
 </p>
 
-### fnml:functionValue
+<p class="issue" data-number="12" data-format="markdown">
+A proper Term map definition in RML is pending.
+For now, we refer to the R2RML spec, but it is assumed these references will be updated based on the evolution of RML.
+</p>
+
+### fnml:Execution
+
+<dfn class="lint-ignore">fnml:Execution</dfn> is a new class to denote an [=FnML execution=].
+It is refered from a [=fnml:ExecutionTermMap=] via the predicate `fnml:execution`.
+It refers to an FnO [=function description=] via the predicate `fnml:function`,
+and to zero or more input parameters via the predicate `fnml:inputParameters`.
 
-<dfn>fnml:functionValue</dfn> connects the knowledge graph generating triples map with a function execution triples map.
-It has range <a>fnml:FunctionMap</a> and domain rr:TriplesMap.
-The triples map to which <a>fnml:functionValue</a> refers to should be an [[RML]] conforming triples map
-that generates an <a>execution</a> description.
+### fnml:ParameterTermMap
+
+<dfn>fnml:ParameterTermMap</dfn> is a subclass of [rr:TermMap](http://www.w3.org/ns/r2rml#TermMap).
```

Comment:
Does it make sense to define this as a `TermMap`? Looking at what the intent is of this construct, it is to generate something along the lines of:
```
_:some-function-execution ex:some-param "some value" .
```

In that sense it is actually closer to a `PredicateObjectMap` which is of course how it was conceived in the original FnML/FnO.

I think we need to decide: Is this approach syntax sugar for the original approach? Or are they actually new constructs.

In the latter case, I would name this construct `fnml:ParameterMap`. We can then decide if this is an intersection of `PredicateObjectMap` and `ObjectMap` or just a special `ExpressionMap`.

Reply:
Hmmm, you're totally right. I tried to refine based on your comments, and the result isn't that far off of the original, see 
![image](https://user-images.githubusercontent.com/668581/156209559-500dc477-b412-4e88-973c-8c997732f1d4.png)

With the added advantage of the original approach that you can dynamically generate function and parameter IRIs. Maybe just replacing rr:TriplesMap in the original approach with a newly introduced concept (e.g. fnml:Execution) and adding the right kind of syntax sugar would be enough.